### PR TITLE
Use "mac_in_use" to add non-vm interface as endpoint

### PR DESCRIPTION
**Pull request by cl1111**
After modification, under the same VLAN, when DHCP service is configured, the VM can obtain the IP address.
image

### CODE REVIEW

**Comment by cl1111** [comment]
fix #382

File: pkg/monitor/agent.go [hunk 1]
```diff
@@ -446,11 +453,20 @@ func (monitor *AgentMonitor) filterEndpointUpdated(rowupdate ovsdb.RowUpdate) (*
 }
 
 func (monitor *AgentMonitor) filterEndpointAdded(rowupdate ovsdb.RowUpdate) *datapath.Endpoint {
+	var macStr string
 	newExternalIds := rowupdate.New.Fields["external_ids"].(ovsdb.OvsMap).GoMap
+	newMacInUse, _ := rowupdate.New.Fields["mac_in_use"].(string)
```
Comment (KeyongSun) [inline pkg/monitor/agent.go:458]: Move this to L461.

File: pkg/monitor/agent.go [hunk 2]
```diff
@@ -407,8 +408,15 @@ func (monitor *AgentMonitor) filterEndpoint(rowupdate ovsdb.RowUpdate) (*datapat
...
 
 func (monitor *AgentMonitor) filterEndpointUpdated(rowupdate ovsdb.RowUpdate) (*datapath.Endpoint, *datapath.Endpoint) {
+	var macStr string
 	newExternalIds := rowupdate.New.Fields["external_ids"].(ovsdb.OvsMap).GoMap
-
+	_, ok := newExternalIds[LocalEndpointIdentity]
+	if !ok {
+		macStr, _ = rowupdate.New.Fields["mac_in_use"].(string)
+	} else {
+		newExternalIds := rowupdate.New.Fields["external_ids"].(ovsdb.OvsMap).GoMap
```
Comment (KeyongSun) [inline pkg/monitor/agent.go:417]: this line is dup as L412

**Review by KeyongSun — COMMENTED** [review]
Add the test case to cover the internal port.

File: pkg/monitor/agent.go (hunk 2 above)
Comment (cl1111) [reply pkg/monitor/agent.go]: Has been modified

File: pkg/monitor/agent.go (hunk 1 above)
Comment (cl1111) [reply pkg/monitor/agent.go]: Has been modified

File: pkg/monitor/agent.go [hunk 3]
```diff
@@ -503,34 +517,36 @@ func (monitor *AgentMonitor) isPortExists(interfaceName string) bool {
...
 	var ofport uint32
-	if rowupdate.Old.Fields["external_ids"] == nil {
-		return nil
-	}
+	var macStr string
 	oldExternalIds := rowupdate.Old.Fields["external_ids"].(ovsdb.OvsMap).GoMap
+	oldMacInUse, _ := rowupdate.Old.Fields["mac_in_use"].(string)
+	_, ok := oldExternalIds[LocalEndpointIdentity]
+	if !ok {
+		macStr = oldMacInUse
```
Comment (KeyongSun) [inline pkg/monitor/agent.go:525]: macStr = rowupdate.Old.Fields["mac_in_use"].(string)
remove L522

Comment (cl1111) [reply pkg/monitor/agent.go]: Has been modified

**Review by KeyongSun — APPROVED** [review]
+1

**Pull request merged**